Index: Doctrine/lib/Doctrine/Migration/Diff.php
===================================================================
--- Doctrine.orig/lib/Doctrine/Migration/Diff.php	2011-03-21 16:08:55.156144000 +0200
+++ Doctrine/lib/Doctrine/Migration/Diff.php	2011-03-21 16:09:20.266144000 +0200
@@ -390,7 +390,11 @@
             }
         } else {
             try {
-                Doctrine_Core::generateModelsFromDb($path, (array) $item, $options);
+                foreach($item as $i)
+                {
+                    $this->_generateModels($prefix, $i);
+                }
+                // Doctrine_Core::generateModelsFromDb($path, (array) $item, $options);
                 return $path;
             } catch (Exception $e) {
                 throw new Doctrine_Migration_Exception('Could not generate models from connection: ' . $e->getMessage());
